sql/core/src/test/resources/sql-tests/inputs/string-functions.sql

```diff
@@ -38,3 +38,11 @@ SELECT substring('Spark SQL', 5, 1);
 SELECT substring('Spark SQL' from 5);
 SELECT substring('Spark SQL' from -3);
 SELECT substring('Spark SQL' from 5 for 1);
+
+-- trim/ltrim/rtrim
```

Comment:
`StringExpressionsSuite` can not test these code paths. So add the test to `string-functions.sql`:
https://github.com/apache/spark/pull/24902/files#diff-2eb730f0723800778689468764627032R656
https://github.com/apache/spark/pull/24902/files#diff-2eb730f0723800778689468764627032R756
https://github.com/apache/spark/pull/24902/files#diff-2eb730f0723800778689468764627032R859